### PR TITLE
Bluetooth: Mesh: Light Control manual override timer

### PR DESCRIPTION
**Pull request by cntlccteam**
1. added manual override time handling
2. changed regulator interval to 0

### CODE REVIEW

File: subsys/bluetooth/mesh/light_ctrl_srv.c [hunk 1]
```diff
@@ -543,17 +546,49 @@ static void reg_step(struct k_work *work)
...
-		light_set(srv, light_to_repr(lvl, LINEAR), REG_INT);
+		light_set(srv, light_to_repr(lvl, LINEAR), 0/*REG_INT*/);
 	}
 }
 #endif
 
+void start_manual_override_timer(struct bt_mesh_light_ctrl_srv *srv)
+{
+	BT_DBG("LC start_manual_override_timer");
+	k_delayed_work_submit(&srv->manual_override_timer,
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/light_ctrl_srv.c:559]: This line has a space at the end. This needs to be removed.

**Review by trond-snekvik — CHANGES_REQUESTED** [review]
Thanks for contributing! 

I have some trouble understanding this manual override mechanism and its purpose. From what I can tell, if a scene that disengages control is recalled, we will turn off the lights and a few seconds later, enable control anyway. If the scene engages control, we're unconditionally turning on the lights and starting the regulator. If we receive an occupancy mode, we'll extend the override timeout. 

This seems like a violation of spec, and I don't think it should be the generic behavior. I'd be interested in knowing what scenario this is addressing, so we can work out a solution that can be part of the framework code.

Our compliance checks are having some issues with the commit message formatting, as it does not follow the [commit guidelines](http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/contribute/index.html#commit-guidelines):
- The commit must have a message body describing the change and why it's needed
- The commit title must start with a module prefix for where the change is made. In this case, `Bluetooth: Mesh:`.
- The commit must have a sign-off line at the end, in the format of `Signed-off-by: Firstname Lastname <myemail@mycompany.com>`. The name and email must match the commit author's name and email. Command line git can generate this line automatically by passing `-s` to `git commit`: `git commit -s`. This is a [Developer Certification of Origin](http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/contribute/index.html#developer-certification-of-origin-dco) policy, and is mandatory.

The merge commit should not be excluded from the pull request. We generally never have merge commits in the git tree in the nRF Connect SDK projects. Commits can be removed through the interactive rebase mechanism described in [step 15 of the contribution workflow documentation](http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/contribute/index.html#contribution-workflow). This step in the workflow documentation should also be followed when updating pull requests.

I have also left a few style related comments that need to be resolved, please take a look.

File: include/bluetooth/mesh/light_ctrl_srv.h [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /*
  * Copyright (c) 2020 Nordic Semiconductor ASA
  *
- * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
```
[the remark this answers is not in the document]
Comment (cntlccteam) [reply include/bluetooth/mesh/light_ctrl_srv.h:4]: ok

File: subsys/bluetooth/mesh/light_ctrl_srv.c [hunk 3]
```diff
@@ -543,17 +546,49 @@ static void reg_step(struct k_work *work)
...
+
+bool is_manual_override_timer_running(struct bt_mesh_light_ctrl_srv *srv)
+{
+	if(k_delayed_work_remaining_get(&srv->manual_override_timer)) {
+		return true;
+	} else {
+		return false;
+	}
+}
+
```
[the remark this answers is not in the document]
Comment (cntlccteam) [reply subsys/bluetooth/mesh/light_ctrl_srv.c]: added bt_mesh_light_ctrl_srv_

File: subsys/bluetooth/mesh/light_ctrl_srv.c [hunk 4]
```diff
@@ -543,17 +546,49 @@ static void reg_step(struct k_work *work)
...
+	k_delayed_work_cancel(&srv->manual_override_timer);
+}
+
+bool is_manual_override_timer_running(struct bt_mesh_light_ctrl_srv *srv)
+{
+	if(k_delayed_work_remaining_get(&srv->manual_override_timer)) {
+		return true;
+	} else {
+		return false;
+	}
```
[the remark this answers is not in the document]
Comment (cntlccteam) [reply subsys/bluetooth/mesh/light_ctrl_srv.c]: ok

File: include/bluetooth/mesh/light_ctrl_srv.h [hunk 5]
```diff
@@ -94,6 +94,10 @@ struct bt_mesh_light_ctrl_srv_cfg {
 	uint32_t fade_standby_manual;
 	/** State-wise light levels */
 	uint16_t light[LIGHT_CTRL_STATE_COUNT];
+	// Manual Override Time (TimeMillisecond with resolution 0.001)
```
Comment (joerchan) [inline include/bluetooth/mesh/light_ctrl_srv.h:97]: Please use `/* */` style commenting.

File: include/bluetooth/mesh/light_ctrl_srv.h [hunk 6]
```diff
@@ -94,6 +94,10 @@ struct bt_mesh_light_ctrl_srv_cfg {
...
+	uint32_t override_time;
+	// Indication if Manual Override Timer is enabled (Boolean)
```
Comment (joerchan) [inline include/bluetooth/mesh/light_ctrl_srv.h:99]: Please use `/* */` style commenting.

**Review by joerchan — COMMENTED** [review]
Hi @cntlccteam. Thank you for contributing.

I think something has gone a bit wrong in your changes here now. In the final diff there is only a change to the header file.
There is also merge commits in the pull request now, these needs to be removed, please use rebase instead.

The commit messages also needs to be updated to follow our commit message guidelines.
For more information about the contribution workflow you can see here:
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/contribute/index.html#contribution-workflow

**Comment by trond-snekvik** [comment]
Closing this. The intended changes are carried out in #4634 and #4629.

**Pull request closed without merging**